apps/newsletters-api/src/app/headers.ts

```diff
@@ -0,0 +1,49 @@
+import type { FastifyReply, FastifyRequest } from 'fastify';
+import { isServingUI } from '../apiDeploymentSettings';
+
+type TtlSettings = {
+	cacheMaxAge: number;
+	surrogateMaxAge: number;
+};
+
+const newsletterTtl: TtlSettings = {
+	cacheMaxAge: 60,
+	surrogateMaxAge: 360,
+};
+
+export const getCacheControl = (
+	req: FastifyRequest,
+): TtlSettings | undefined => {
+	// the API instance serving the UI must always provide fresh data
+	if (isServingUI()) {
+		return undefined;
+	}
+
+	if (req.routerPath.startsWith('/api/newsletters')) {
+		return newsletterTtl;
+	}
+	if (req.routerPath.startsWith('/api/legacy')) {
+		return newsletterTtl;
+	}
+
+	return undefined;
+};
+
+export const setHeaderHook = async (
+	req: FastifyRequest,
+	reply: FastifyReply,
+) => {
+	const cacheControl = getCacheControl(req);
+	if (!cacheControl) {
+		return;
+	}
+
+	// Fastly gives priority to the Surrogate-Control header to determine how often to
```

Comment:
Is there any need for the complexity of the split Fastly/Browser cache control headers in this case?

This is often useful when you are making use of Fastly's cache purge APIs - so you can set a much longer surrogate TTL than cache-control TTL, safe in the knowledge that the cache purge API will honour your request to purge regardless of the headers.

But I don't think this service makes use of Fastly's cache purge APIs, so just setting cache-control headers should be sufficient.

Reply:
Probably not - my initial reading of https://docs.fastly.com/en/guides/caching-best-practices#understand-how-cache-control-headers-work was to using surrogate control was recommended generally, but I think I got that wrong.